src/utils.js

```diff
@@ -1,9 +1,12 @@
+const hasOwn = Object.prototype.hasOwnProperty
+
+const EMPTY = []
 const SPLITTER = '.'
 export function castPath (value) {
-  if (value == null) {
-    return []
-  } else if (Array.isArray(value)) {
+  if (Array.isArray(value)) {
     return value
+  } else if (value === '' || value === null || value === undefined) {
+    return EMPTY
   }
```

Comment:
This looks pedantic, but it grants V8 sufficient type information to give this function a pretty high level of optimization. Making this change roughly doubled the speed of the `push` performance benchmark.